### PR TITLE
Remove minimum constraint on OpenSearch hosts to allow empty host and refactor status API to not include OpenSearch status when host is empty

### PR DESCRIPTION
**Pull request by bandinib-amzn**
### Description

- Remove minimum constraint on OpenSearch hosts to allow empty host. 
- Bypass node compatibility check when empty opensearch host is configured.
- 'Opensearch is not configured' message in status API when empty opensearch host is configured.

Please note no opensearch host is only considered when explicitly configured. That means you need to add `opensearch.hosts: []` in `opensearch_dashboards.yml` file. Default value will remain same which is `http://localhost:9200` in case `opensearch.hosts` is commented in `opensearch_dashboards.yml` file.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4700

## Screenshot

Status page when `#opensearch.hosts: ["http://localhost:9200"]` OR `opensearch.hosts: ["http://localhost:9200"]` in `opensearch_dashboards.yml` file.
Default Status Page

Status page when `opensearch.hosts:[]` in `opensearch_dashboards.yml` file.

Screenshot 2023-08-22 at 10 00 35 AM

API Status response:
```
% curl localhost:5601/api/status | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 10279  100 10279    0     0  2987k      0 --:--:-- --:--:-- --:--:-- 3346k
{
  "name": "dev-dsk-bandinib-2c-1b8c3fb7.us-west-2.amazon.com",
  "uuid": "9e48a33a-9b44-4488-9cdc-22244cb1c8af",
  "version": {
    "number": "3.0.0",
    "build_hash": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "build_number": 9007199254740991,
    "build_snapshot": false
  },
  "status": {
    "overall": {
      "since": "2023-08-22T16:58:09.161Z",
      "state": "green",
      "title": "Green",
      "nickname": "Looking good",
      "icon": "success",
      "uiColor": "secondary"
    },
    "statuses": [
      {
        "id": "core:opensearch@3.0.0",
        "message": "Opensearch is not configured",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "core:savedObjects@3.0.0",
        "message": "Saved Object Service is using Postgres and it is up",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:advancedSettings@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:bfetch@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:charts@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:dataSourceManagement@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:devTools@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:discover@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:embeddable@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:expressions@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:indexPatternManagement@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:inspector@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:legacyExport@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:management@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:managementOverview@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:navigation@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchDashboardsOverview@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchDashboardsReact@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchDashboardsUsageCollection@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchDashboardsUtils@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchUiShared@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:savedObjects@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:savedObjectsManagement@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:share@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:uiActions@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:urlForwarding@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visDefaultEditor@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visualizations@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visualize@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:dashboardsStoragePlugin@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:apmOss@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:console@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:dashboard@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:data@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:dataSource@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:home@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:inputControlVis@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:mapsLegacy@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:opensearchDashboardsLegacy@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:regionMap@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:tileMap@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:usageCollection@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visAugmenter@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visBuilder@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeMarkdown@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeMetric@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeTable@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeTagcloud@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeTimeline@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeTimeseries@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeVega@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      },
      {
        "id": "plugin:visTypeVislib@3.0.0",
        "message": "All dependencies are available",
        "since": "2023-08-22T16:58:09.161Z",
        "state": "green",
        "icon": "success",
        "uiColor": "secondary"
      }
    ]
  },
  "metrics": {
    "last_updated": "2023-08-22T16:58:07.854Z",
    "collection_interval_in_millis": 5000,
    "os": {
      "platform": "linux",
      "platformRelease": "linux-5.4.250-173.369.amzn2int.x86_64",
      "load": {
        "1m": 0.09,
        "5m": 0.11,
        "15m": 0.05
      },
      "memory": {
        "total_in_bytes": 16170184704,
        "free_in_bytes": 14283046912,
        "used_in_bytes": 1887137792
      },
      "uptime_in_millis": 134766270,
      "distro": "Amazon Linux",
      "distroRelease": "Amazon Linux-2 (Karoo)",
      "cpuacct": {
        "control_group": "/user.slice",
        "usage_nanos": 9161043095779
      },
      "cpu": {
        "control_group": "/user.slice",
        "cfs_period_micros": 100000,
        "cfs_quota_micros": -1,
        "stat": {
          "number_of_elapsed_periods": 0,
          "number_of_times_throttled": 0,
          "time_throttled_nanos": 0
        }
      }
    },
    "process": {
      "memory": {
        "heap": {
          "total_in_bytes": 127447040,
          "used_in_bytes": 110396848,
          "size_limit": 2197815296
        },
        "resident_set_size_in_bytes": 252866560
      },
      "pid": 19229,
      "event_loop_delay": 0.3745400011539459,
      "uptime_in_millis": 88092.057665
    },
    "response_times": {
      "avg_in_millis": 0,
      "max_in_millis": 0
    },
    "concurrent_connections": 6,
    "requests": {
      "disconnects": 0,
      "total": 0,
      "statusCodes": {},
      "status_codes": {}
    }
  }
}
```
### Check List

- [X] All tests pass
  - [X] `yarn test:jest`
  - [X] `yarn test:jest_integration`
  - [X] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by zhongnansu — DISMISSED** [review]
lgtm

File: src/core/server/opensearch/status.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const calculateStatus$ = (
 
           return {
             level: ServiceStatusLevels.available,
-            summary: `OpenSearch is available`,
+            summary: message ?? `OpenSearch is available`,
```
Comment (AMoo-Miki) [inline src/core/server/opensearch/status.ts:80]:
```suggestion
            summary: message || `OpenSearch is available`,
```
`||` is a safer operator to guard against `''` and `false`.

Comment (AMoo-Miki) [reply src/core/server/opensearch/status.ts:80]: Though, i would prefer to be stricter..
```suggestion
            summary: (message ?? `OpenSearch is available`) || `Unknown`,
```

**Review by AMoo-Miki — DISMISSED** [review]

File: src/core/server/opensearch/status.ts (hunk 1 above)
Comment (bandinib-amzn) [reply src/core/server/opensearch/status.ts]: Good idea! Refactored to be more stricter.

**Review by AMoo-Miki — APPROVED** [review]

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**